### PR TITLE
Fix: 🐛 Contradicting argument configurations

### PR DESCRIPTION
**Pull request by KP64**
Right now, there are the following output coloring options:
```rs
 /// Turn on colorization always
#[arg(short = 'C', long)]
pub force_color: bool,
```
and
```rs
/// Print plainly without ANSI escapes
#[arg(long)]
pub no_color: bool,
```
This allows Users to contradict themselves when passing arguments like this:
```
cargo run -- --no-color --force-color // It will force the color btw
```
Which is why I've come with the following solution:
```rs
#[derive(Clone, Copy, Debug, clap::ValueEnum, PartialEq, Eq, Default)]
pub enum Coloring {
    None,
    #[default]
    Colored,
    Forced,
}
```
and
```rs
#[arg(short = 'C', long, value_enum, default_value_t = Coloring::default())]
pub coloring: Coloring,
```
This also allows Users to specify the coloring output they want with “-C” alone, making it more intuitive.

### CODE REVIEW

**Review by solidiquis — CHANGES_REQUESTED** [review]
This is much better. Thank you! Could you also update the `usage` section of the README to reflect the new arguments? I can update [this part of the readme](https://github.com/solidiquis/erdtree#redirecting-output-and-colorization) later once this is merged.

**Comment by ilyapopov** [comment]
I would suggest "auto" instead of "colored". Maybe also add a long alias `--color`.

File: src/render/context/mod.rs
Comment (solidiquis) [inline src/render/context/mod.rs:61]: Final change: Would you mind moving the doc comments for each enum variant over to the enum itself? `clap` knows how to handle the `--help` output when done in that fashion. Checkout `src/render/context/sort.rs` as a reference

**Review by solidiquis — CHANGES_REQUESTED** [review]

File: src/render/context/mod.rs
Comment (solidiquis) [reply src/render/context/mod.rs]: For someone new to Rust you have fantastic intuition. This is fine! I'll approve and merge this once tests pass. If I need to make any small adjustments to the documentation I can do that later. Thanks a bunch for the contribution!

Comment (solidiquis) [reply src/render/context/mod.rs]: If I don't merge this within the next few minutes I'll merge it immediately after I get off work.

**Review by solidiquis — APPROVED** [review]

**Pull request merged**
